src/RawPOSIX/src/constants/fs_constants.rs

```diff
@@ -0,0 +1,192 @@
+pub const STDIN_FILENO: i32 = 0;    // File descriptor for standard input
+pub const STDOUT_FILENO: i32 = 1;   // File descriptor for standard output
+pub const STDERR_FILENO: i32 = 2;   // File descriptor for standard error
+
+// ===== File Descriptor Constants =====
+pub const DT_UNKNOWN: u8 = 0;
+
+pub const STARTINGFD: i32 = 0;       // Starting file descriptor number
+pub const MAXFD: i32 = 1024;         // Maximum number of file descriptors
+pub const STARTINGPIPE: i32 = 0;     // Starting pipe descriptor number
+pub const MAXPIPE: i32 = 1024;       // Maximum number of pipes
+
+// ===== Inode Constants =====
+pub const ROOTDIRECTORYINODE: usize = 1;  // Root directory inode number
+pub const STREAMINODE: usize = 2;         // Stream inode number
+
+// ===== Pipe Constants =====
+pub const PIPE_CAPACITY: usize = 65536;   // Maximum pipe buffer size
+
+// ===== File Access Permission Flags =====
+pub const F_OK: u32 = 0;     // Test for existence
+pub const X_OK: u32 = 1;     // Test for execute permission
+pub const W_OK: u32 = 2;     // Test for write permission
+pub const R_OK: u32 = 4;     // Test for read permission
+
+// ===== File Access Modes =====
+// Source: include/uapi/asm-generic/fcntl.h
+pub const O_RDONLY: i32 = 0o0;       // Open read-only
+pub const O_WRONLY: i32 = 0o1;       // Open write-only
+pub const O_RDWR: i32 = 0o2;         // Open read-write
+pub const O_RDWRFLAGS: i32 = 0o3;    // Mask for access modes
+
+// ===== File Creation and Status Flags =====
+pub const O_CREAT: i32 = 0o100;      // Create file if it doesn't exist
+pub const O_EXCL: i32 = 0o200;       // Error if O_CREAT and file exists
+pub const O_NOCTTY: i32 = 0o400;     // Don't assign controlling terminal
+pub const O_TRUNC: i32 = 0o1000;     // Truncate file to zero length
+pub const O_APPEND: i32 = 0o2000;    // Append mode - writes always at end
+pub const O_NONBLOCK: i32 = 0o4000;  // Non-blocking mode
+pub const O_SYNC: i32 = 0o10000;     // Synchronous writes
+pub const O_ASYNC: i32 = 0o20000;    // Signal-driven I/O
+pub const O_CLOEXEC: i32 = 0o2000000; // Close on exec
+
+pub const DEFAULTTIME: u64 = 1323630836;  // Default timestamp value
+
+// ===== File Permissions =====
+// Source: include/uapi/linux/stat.h
+pub const S_IRWXA: u32 = 0o777;  // All permissions for all users
+pub const S_IRWXU: u32 = 0o700;  // User read, write, execute
+pub const S_IRUSR: u32 = 0o400;  // User read
+pub const S_IWUSR: u32 = 0o200;  // User write
+pub const S_IXUSR: u32 = 0o100;  // User execute
+pub const S_IRWXG: u32 = 0o070;  // Group read, write, execute
+pub const S_IRGRP: u32 = 0o040;  // Group read
+pub const S_IWGRP: u32 = 0o020;  // Group write
+pub const S_IXGRP: u32 = 0o010;  // Group execute
+pub const S_IRWXO: u32 = 0o007;  // Others read, write, execute
+pub const S_IROTH: u32 = 0o004;  // Others read
+pub const S_IWOTH: u32 = 0o002;  // Others write
+pub const S_IXOTH: u32 = 0o001;  // Others execute
+
+//Commands for FCNTL
```

Comment:
At least point to the man pages for all of this.   Also, if anything changes from Linux, definitely say something.